### PR TITLE
Fuse warning

### PR DESCRIPTION
**Pull request by jdoliner**
This PR has a few small cosmetic fixes for FUSE:

- prints a warning on macOS since we've observed some instability there, and we know it will fail on new versions.
- changes the names we pass to the mount syscall so mounted filesystems will show up as having type pfs and name fuse.pfs.

It also makes a few minor non-cosmetic changes to `unmount`:

- Fixes the `--all` flag, this was broken for a while due to not having the above change that lists our mounts as having type pfs.
- Prints an error if you do unmount without an argument or the `--all` flag. This was a noop previously, but that can be a bit confusing so I made it error.

### CODE REVIEW

**Comment by nitinjainsj** [comment]
@jdoliner 
A few questions
- Do we tell users that mount is FUSE-based or is this the first time many of them will learn?
- Is there a mac version where FUSE is officially supported?
- Why do we need to show the warning on Linux?

Suggestion... "pachctl mount is supported on MacOS version XYZ or earlier"

**Comment by jdoliner** [comment]
We mention that it's FUSE backed in docs I think. But the command itself is called `mount` so that kind of obfuscates it.

Afaik I know there's never been a version of macOS that officially supported FUSE. Although older versions have definitely had better community support than the current version.

We should not be showing the error on Linux, that was a copy-paste error on my part :sweat_smile:

I like your suggest message, I'm just not quite sure what version I should say it was officially supported on.

**Comment by nitinjainsj** [comment]
Maybe the message should be "pachctl mount is supported on MacOS version 1.10.15 or earlier."

File: src/server/pfs/cmds/mount_darwin.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// +build darwin
+
+package cmds
+
+import "fmt"
+
+func printWarning() {
+	fmt.Println(`WARNING: Mount is supported on macOS versions 1.10.5 and earlier. Mount is implemented using FUSE which isn't supported in macOS 1.11. FUSE on macOS differs from Linux FUSE which causes some issues known issues. For the best experience we recommend using mount on Linux.`)
+}
```
Comment (nitinjainsj) [inline src/server/pfs/cmds/mount_darwin.go:9]: Great. This is much clear and proposes alternatives.

**Pull request merged**